From 05273f7bc4f3b02de79df360262834fb26be0dd0 Mon Sep 17 00:00:00 2001
From: GuaiYiHu <guaiyihu@foxmail.com>
Date: Wed, 7 Oct 2026 21:05:39 +0800
Subject: [PATCH] soong: Pass the optional device non-PIE allowlist directory

Export LINKER_NON_PIE_EXECUTABLES_HEADER_DIR to the linker product-variable properties. Use JSON null when unset so builds without a device allowlist do not compile the optional compatibility code or require its header.

Change-Id: I3d2cdd3ba6eb486c328f7fa8e2aa38cd423da524
Signed-off-by: GuaiYiHu <guaiyihu@foxmail.com>
---
 build/soong/android/variable.go | 6 ++++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 7 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index 2136f01e..311f9bb2 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -1,5 +1,10 @@
 package android
 type Product_variables struct {
+	Non_pie_executables_header_dir struct {
+		Cppflags []string
+		Include_dirs []string
+		Srcs []string
+	}
 	Additional_gralloc_10_usage_bits struct {
 		Cppflags []string
 	}
@@ -74,6 +79,7 @@ type Product_variables struct {
 }
 
 type ProductVariables struct {
+	Non_pie_executables_header_dir *string `json:",omitempty"`
 	Additional_gralloc_10_usage_bits  *string `json:",omitempty"`
 	Apply_msm8974_1440p_egl_workaround  *bool `json:",omitempty"`
 	Bootloader_message_offset  *int `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 0820d67b..9b5a51e2 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -4,6 +4,7 @@ add_json_val_default = $(call add_json_val, $(1), $(if $(strip $(2)), $(2), $(3)
 _contents := $(_contents)    "Lineage":{$(newline)
 
 # See build/core/soong_config.mk for the add_json_* functions you can use here.
+$(call add_json_str_omitempty, Non_pie_executables_header_dir, $(LINKER_NON_PIE_EXECUTABLES_HEADER_DIR))
 $(call add_json_str_omitempty, Additional_gralloc_10_usage_bits, $(TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS))
 $(call add_json_bool, Apply_msm8974_1440p_egl_workaround, $(filter true,$(TARGET_MSM8974_1440P_EGL_WORKAROUND)))
 $(call add_json_val_default, Bootloader_message_offset, $(BOOTLOADER_MESSAGE_OFFSET), 0)
